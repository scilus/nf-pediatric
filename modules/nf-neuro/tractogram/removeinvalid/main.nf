process TRACTOGRAM_REMOVEINVALID {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
        tuple val(meta), path(tractogram)

    output:
        tuple val(meta), path("*.{trk,tck}"), emit: tractograms
        path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "_${task.ext.suffix}" : ""

    def force = task.ext.force ? "-f" : ""
    def cut_invalid = task.ext.cut_invalid ? "--cut_invalid" : ""
    def remove_single_point = task.ext.remove_single_point ? "--remove_single_point" : ""
    def remove_overlapping_points = task.ext.remove_overlapping_points ? "--remove_overlapping_points" : ""
    def threshold = task.ext.threshold ? "--threshold " + task.ext.threshold : ""
    def no_empty = task.ext.no_empty ? "--no_empty" : ""

    """
    for tractogram in ${tractogram};
        do \
        ext=\${tractogram#*.}
        bname=\$(basename \${tractogram} .\${ext})

        scil_tractogram_remove_invalid.py \$tractogram ${prefix}__\${bname}${suffix}.\${ext}\
                        $cut_invalid\
                        $remove_single_point\
                        $remove_overlapping_points\
                        $threshold\
                        $no_empty\
                        $force

    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def suffix = task.ext.suffix ? "_${task.ext.suffix}" : ""

    """
    scil_tractogram_remove_invalid.py -h

    for tractogram in ${tractogram};
        do \
        ext=\${tractogram#*.}
        bname=\$(basename \${tractogram} .\${ext})

        touch ${prefix}__\${bname}${suffix}.\${ext}
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """
}
