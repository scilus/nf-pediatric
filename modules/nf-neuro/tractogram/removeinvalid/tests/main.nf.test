nextflow_process {

    name "Test Process TRACTOGRAM_REMOVEINVALID"
    script "../main.nf"
    process "TRACTOGRAM_REMOVEINVALID"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "tractogram"
    tag "tractogram/removeinvalid"

    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "bundles.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("tractogram - remove invalid") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/fibercup_atlas/subj_1/*.trk", checkIfExists: true),
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
