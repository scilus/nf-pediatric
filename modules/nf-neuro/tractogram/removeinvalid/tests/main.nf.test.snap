{
    "tractogram - remove invalid": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__bundle_0.trk:md5,6fb44e5444b09eaf6e612e1bde04b0ed",
                            "test__bundle_1.trk:md5,04fbc7404a45e197d203d05e1bde351a",
                            "test__bundle_2.trk:md5,91cf2d5b1e8ca0e219012dbcaf4a8edb",
                            "test__bundle_3.trk:md5,a2b56e397fc062691ed4c99f8fe67228",
                            "test__bundle_4.trk:md5,3fe0cd679a23e53079bcfdfb3a777e89",
                            "test__bundle_5.trk:md5,9a7d5da64e557eb6622a051593886e4a",
                            "test__bundle_6.trk:md5,29434c0aa3bc72c721e8879edd4f0dca"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,9c98a9977d211116b5e0585b635769c1"
                ],
                "tractograms": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__bundle_0.trk:md5,6fb44e5444b09eaf6e612e1bde04b0ed",
                            "test__bundle_1.trk:md5,04fbc7404a45e197d203d05e1bde351a",
                            "test__bundle_2.trk:md5,91cf2d5b1e8ca0e219012dbcaf4a8edb",
                            "test__bundle_3.trk:md5,a2b56e397fc062691ed4c99f8fe67228",
                            "test__bundle_4.trk:md5,3fe0cd679a23e53079bcfdfb3a777e89",
                            "test__bundle_5.trk:md5,9a7d5da64e557eb6622a051593886e4a",
                            "test__bundle_6.trk:md5,29434c0aa3bc72c721e8879edd4f0dca"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9c98a9977d211116b5e0585b635769c1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-21T14:39:05.238848"
    }
}